internal/datanode/writebuffer/delta_buffer.go

```diff
@@ -0,0 +1,80 @@
+package writebuffer
+
+import (
+	"math"
+
+	"github.com/milvus-io/milvus-proto/go-api/v2/msgpb"
+	"github.com/milvus-io/milvus-proto/go-api/v2/schemapb"
+	"github.com/milvus-io/milvus/internal/storage"
+	"github.com/milvus-io/milvus/pkg/util/paramtable"
+	"github.com/milvus-io/milvus/pkg/util/typeutil"
+)
+
+type DeltaBuffer struct {
```

Comment:
does it need a mutex？

Reply:
Delta buffer is protected inside `WriteBuffer`. There shall be no direct operation outside of this package